File: Components/Document Server/CS/DocsMainDemo/Modules/PdfPasswordProtection.cs
using System;
using System.Security;
using DevExpress.Pdf;
using DevExpress.DXperience.Demos;
using DevExpress.XtraEditors.Controls;

namespace DevExpress.Docs.Demos {
    public partial class PdfPasswordProtection : TutorialControlBase {
        static readonly string path = DemoUtils.GetRelativePath("EncryptDemo.pdf");

        public override bool NoGap { get { return true; } }

        public PdfPasswordProtection() {
            InitializeComponent();
            imageComboBoxPrinting.Properties.Items.Add(new ImageComboBoxItem("Not allowed", PdfDocumentPrintingPermissions.NotAllowed, 0));
            imageComboBoxPrinting.Properties.Items.Add(new ImageComboBoxItem("Low quality", PdfDocumentPrintingPermissions.LowQuality, 0));
            imageComboBoxPrinting.Properties.Items.Add(new ImageComboBoxItem("Allowed", PdfDocumentPrintingPermissions.Allowed, 0));
            imageComboBoxPrinting.SelectedIndex = 2;
            imageComboBoxDataExtraction.Properties.Items.Add(new ImageComboBoxItem("Not allowed", PdfDocumentDataExtractionPermissions.NotAllowed, 0));
            imageComboBoxDataExtraction.Properties.Items.Add(new ImageComboBoxItem("Accessibility", PdfDocumentDataExtractionPermissions.Accessibility, 0));
            imageComboBoxDataExtraction.Properties.Items.Add(new ImageComboBoxItem("Allowed", PdfDocumentDataExtractionPermissions.Allowed, 0));
            imageComboBoxDataExtraction.SelectedIndex = 2;
            imageComboBoxModifying.Properties.Items.Add(new ImageComboBoxItem("Not allowed", PdfDocumentModificationPermissions.NotAllowed, 0));
            imageComboBoxModifying.Properties.Items.Add(new ImageComboBoxItem("Document assembling", PdfDocumentModificationPermissions.DocumentAssembling, 0));
            imageComboBoxModifying.Properties.Items.Add(new ImageComboBoxItem("Allowed", PdfDocumentModificationPermissions.Allowed, 0));
            imageComboBoxModifying.SelectedIndex = 2;
            imageComboBoxInteractivity.Properties.Items.Add(new ImageComboBoxItem("Not allowed", PdfDocumentInteractivityPermissions.NotAllowed, 0));
            imageComboBoxInteractivity.Properties.Items.Add(new ImageComboBoxItem("Form filling and signing", PdfDocumentInteractivityPermissions.FormFillingAndSigning, 0));
            imageComboBoxInteractivity.Properties.Items.Add(new ImageComboBoxItem("Allowed", PdfDocumentInteractivityPermissions.Allowed, 0));
            imageComboBoxInteractivity.SelectedIndex = 2;
            Enabled = PdfFileHelper.LoadDocument(pdfViewer, path);
        }
        void OnButtonEncryptClick(object sender, EventArgs e) {
            using (PdfDocumentProcessor documentProcessor = new PdfDocumentProcessor()) {
                documentProcessor.LoadDocument(path);
                using (SecureString ownerPassword = new SecureString()) {
                    foreach (char c in teOwnerPassword.Text)
                        ownerPassword.AppendChar(c);
                    using (SecureString userPassword = new SecureString()) {
                        foreach (char c in teUserPassword.Text)
                            userPassword.AppendChar(c);
                        PdfEncryptionOptions encryptionOptions = new PdfEncryptionOptions();
                        encryptionOptions.PrintingPermissions = (PdfDocumentPrintingPermissions)imageComboBoxPrinting.EditValue;
                        encryptionOptions.DataExtractionPermissions = (PdfDocumentDataExtractionPermissions)imageComboBoxDataExtraction.EditValue;
                        encryptionOptions.ModificationPermissions = (PdfDocumentModificationPermissions)imageComboBoxModifying.EditValue;
                        encryptionOptions.InteractivityPermissions = (PdfDocumentInteractivityPermissions)imageComboBoxInteractivity.EditValue;
                        encryptionOptions.OwnerPassword = ownerPassword;
                        encryptionOptions.UserPassword = userPassword;
                        PdfFileHelper.Save(documentProcessor, new PdfSaveOptions() { EncryptionOptions = encryptionOptions });
                    }
                }
            }
        }
    }
}

File: Components/Document Server/VB/DocsMainDemo/Modules/PdfPasswordProtection.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Security
Imports DevExpress.Pdf
Imports DevExpress.DXperience.Demos
Imports DevExpress.XtraEditors.Controls

Namespace DevExpress.Docs.Demos
	Partial Public Class PdfPasswordProtection
		Inherits TutorialControlBase
		Private Shared ReadOnly path As String = DemoUtils.GetRelativePath("EncryptDemo.pdf")

		Public Overrides ReadOnly Property NoGap() As Boolean
			Get
				Return True
			End Get
		End Property

		Public Sub New()
			InitializeComponent()
			imageComboBoxPrinting.Properties.Items.Add(New ImageComboBoxItem("Not allowed", PdfDocumentPrintingPermissions.NotAllowed, 0))
			imageComboBoxPrinting.Properties.Items.Add(New ImageComboBoxItem("Low quality", PdfDocumentPrintingPermissions.LowQuality, 0))
			imageComboBoxPrinting.Properties.Items.Add(New ImageComboBoxItem("Allowed", PdfDocumentPrintingPermissions.Allowed, 0))
			imageComboBoxPrinting.SelectedIndex = 2
			imageComboBoxDataExtraction.Properties.Items.Add(New ImageComboBoxItem("Not allowed", PdfDocumentDataExtractionPermissions.NotAllowed, 0))
			imageComboBoxDataExtraction.Properties.Items.Add(New ImageComboBoxItem("Accessibility", PdfDocumentDataExtractionPermissions.Accessibility, 0))
			imageComboBoxDataExtraction.Properties.Items.Add(New ImageComboBoxItem("Allowed", PdfDocumentDataExtractionPermissions.Allowed, 0))
			imageComboBoxDataExtraction.SelectedIndex = 2
			imageComboBoxModifying.Properties.Items.Add(New ImageComboBoxItem("Not allowed", PdfDocumentModificationPermissions.NotAllowed, 0))
			imageComboBoxModifying.Properties.Items.Add(New ImageComboBoxItem("Document assembling", PdfDocumentModificationPermissions.DocumentAssembling, 0))
			imageComboBoxModifying.Properties.Items.Add(New ImageComboBoxItem("Allowed", PdfDocumentModificationPermissions.Allowed, 0))
			imageComboBoxModifying.SelectedIndex = 2
			imageComboBoxInteractivity.Properties.Items.Add(New ImageComboBoxItem("Not allowed", PdfDocumentInteractivityPermissions.NotAllowed, 0))
			imageComboBoxInteractivity.Properties.Items.Add(New ImageComboBoxItem("Form filling and signing", PdfDocumentInteractivityPermissions.FormFillingAndSigning, 0))
			imageComboBoxInteractivity.Properties.Items.Add(New ImageComboBoxItem("Allowed", PdfDocumentInteractivityPermissions.Allowed, 0))
			imageComboBoxInteractivity.SelectedIndex = 2
			Enabled = PdfFileHelper.LoadDocument(pdfViewer, path)
		End Sub
		Private Sub OnButtonEncryptClick(ByVal sender As Object, ByVal e As EventArgs) Handles buttonEncrypt.Click
			Using documentProcessor As New PdfDocumentProcessor()
				documentProcessor.LoadDocument(path)
				Using ownerPassword As New SecureString()
					For Each c As Char In teOwnerPassword.Text
						ownerPassword.AppendChar(c)
					Next c
					Using userPassword As New SecureString()
						For Each c As Char In teUserPassword.Text
							userPassword.AppendChar(c)
						Next c
						Dim encryptionOptions As New PdfEncryptionOptions()
						encryptionOptions.PrintingPermissions = CType(imageComboBoxPrinting.EditValue, PdfDocumentPrintingPermissions)
						encryptionOptions.DataExtractionPermissions = CType(imageComboBoxDataExtraction.EditValue, PdfDocumentDataExtractionPermissions)
						encryptionOptions.ModificationPermissions = CType(imageComboBoxModifying.EditValue, PdfDocumentModificationPermissions)
						encryptionOptions.InteractivityPermissions = CType(imageComboBoxInteractivity.EditValue, PdfDocumentInteractivityPermissions)
						encryptionOptions.OwnerPassword = ownerPassword
						encryptionOptions.UserPassword = userPassword
						PdfFileHelper.Save(documentProcessor, New PdfSaveOptions() With {.EncryptionOptions = encryptionOptions})
					End Using
				End Using
			End Using
		End Sub
	End Class
End Namespace
